### ISSUE TITLE
The trainable_variables of the subclass of tf.keras.Model is empty.

### ISSUE THREAD

**Issue by over-shine** [tensorflow/tensorflow#39358]
class MaskMean(tf.keras.layers.Layer):
    def __init__(self, trainable=True):
        super(MaskMean, self).__init__()
        self.dropout = tf.keras.layers.Dropout(0.1, trainable=trainable)

class Predict(tf.keras.layers.Layer):
    def __init__(self, trainable=True):
        super(Predict, self).__init__()
        self.dense_layer = tf.keras.layers.Dense(512, activation='relu', name="full connection", trainable=trainable)
        self.predict = tf.keras.layers.Dense(1, name="prediction", trainable=trainable)

class Regulation(tf.keras.Model):
    def __init__(self, trainable=True):
        super(Regulation, self).__init__()
        # self.embedding = TFBertModel.from_pretrained(r'E:\googleCloudDisk\model\Bert\cased\base')
        # self.dropout = tf.keras.layers.Dropout(0.1, name='dropout')
        # self.inmediate = tf.keras.layers.Dense(512, name="full connection")
        # self.predict = tf.keras.layers.Dense(1, name="prediction")
        self.mask_mean_layer = MaskMean(trainable=trainable)
        self.predict = Predict(trainable=trainable)

Rmodel = Regulation(trainable=True)

But the Rmodel.trainable_variables() is empty.
How to define a subclass model of tf.keras.Model to make its' variables can be trainable and could return by Rmodel.trainable_variables()?

**Comment by ravikyram** [comment]
Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version?.

Request you to share colab link or simple standalone code with proper indentation to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by over-shine** [comment]
I have solved this issue, thanks

**Comment by vsomnath** [comment]
How did you fix this issue?

**Comment by over-shine** [comment]
one batch data for the subclass and then the trainable variables will be got发自我的华为手机-------- 原始邮件 --------发件人： Vignesh Ram Somnath <notifications@github.com>日期： 2020年6月21日周日 17:21收件人： tensorflow/tensorflow <tensorflow@noreply.github.com>抄送： over-shine <Chengwei_works@163.com>, State change <state_change@noreply.github.com>主    题： Re: [tensorflow/tensorflow] The trainable_variables of the subclass of tf.keras.Model is empty. (#39358)
How did you fix this issue?
